cmd/dmverity-vhd/main.go

```diff
@@ -171,70 +171,87 @@ var createVHDCommand = cli.Command{
 
 		layers, err := fetchImageLayers(ctx)
 		if err != nil {
-			return errors.Wrap(err, "failed to fetch image layers")
+			return fmt.Errorf("failed to fetch image layers: %w", err)
 		}
 
 		outDir := ctx.String(outputDirFlag)
 		if _, err := os.Stat(outDir); os.IsNotExist(err) {
 			log.Debugf("creating output directory %q", outDir)
 			if err := os.MkdirAll(outDir, 0755); err != nil {
-				return errors.Wrapf(err, "failed to create output directory %s", outDir)
+				return fmt.Errorf("failed to create output directory %s: %w", outDir, err)
 			}
 		}
 
-		log.Debug("creating layer VHDs with dm-verity:")
+		log.Debug("creating layer VHDs with dm-verity")
 		for layerNumber, layer := range layers {
-			diffID, err := layer.DiffID()
-			if err != nil {
-				return errors.Wrap(err, "failed to read layer diff")
-			}
-			log.Debugf("Layer #%d, layer hash: %s", layerNumber, diffID.String())
-
-			rc, err := layer.Uncompressed()
-			if err != nil {
-				return errors.Wrapf(err, "failed to uncompress layer %s", diffID.String())
-			}
-
-			vhdPath := filepath.Join(ctx.String(outputDirFlag), diffID.Hex+".vhd")
-			out, err := os.Create(vhdPath)
+			diff, err := layer.DiffID()
```

Comment:
this is also computed in `createVHD` too, right?
is there a need for it to be called twice?

Reply:
ACC CLI parses the output I believe, so I didn't want to make substantial changes to what's being logged. since `createVHD` doesn't really need to know about layer index, I left it out.